src/PropertySet.cc

```diff
@@ -202,12 +203,12 @@ T PropertySet::get(std::string const& name)
         throw LSST_EXCEPT(pex::exceptions::NotFoundError, name + " not found");
     }
     try {
-        return boost::any_cast<T>(i->second->back());
-    } catch (boost::bad_any_cast) {
+        return std::any_cast<T>(i->second->back());
+    } catch (std::bad_any_cast &) {
         throw LSST_EXCEPT(pex::exceptions::TypeError, name);
```

Comment:
Not related to the migration, but it might be worth adding exception chaining, to keep information about the badly cast value from being lost:
```suggestion
        std::throw_with_nested(LSST_EXCEPT(pex::exceptions::TypeError, name));
```

Same with other cases below.